apps/music/js/endpoint.js

```diff
@@ -497,4 +497,24 @@ function open(blob) {
   });
 }
 
+function throttle(fn, ms) {
+  var last = Date.now();
+  var timeout;
+  return (...args) => {
+    var now = Date.now();
+    if (now < last + ms) {
+      clearTimeout(timeout);
+      timeout = setTimeout(() => {
+        last = now;
+        fn.apply(this, args);
+      }, ms);
+    }
+
+    else {
+      last = now;
+      fn.apply(this, args);
+    }
+  };
+}
```

Comment:
Yes. I was thinking this when I posted this patch. There are more things than just this that we can probably consolidate though, so I'm gonna hold off and just file another "cleanup" bug to DRY things up a bit :-)